Sources/Identity/CustomerInfo+ActiveDates.swift

```diff
@@ -79,6 +79,17 @@ extension CustomerInfo {
 
 private extension CustomerInfo {
 
+    static func productID(productID: String, purchase: CustomerInfoResponse.Subscription) -> String {
```

Comment:
I'm a bit late to the game, but I think a more swift-y name for this would be something like: 
```swift
static func productID(_ productID: String, from purchase: CustomerInfoResponse.Subscription)
```
and given that this is called from a function called `extractPurchaseDates`, I'd follow the same scheme here, i.e.: 

```swift
static func extractProductIDAndBasePlan(from productID: String, purchase: CustomerInfoResponse.Subscription)
```

- making clear that it's also doing something with a base plan, since having a function called productID that takes the productID seems confusing unless you look at the implementation
- using `extract` for consistency
- using parameter names to make the name slightly cleaner

Reply:
Done in #2708